File: internal/content/testdata/bench/csharp/nc-container-most-water.cs
public class Solution {
    public long max_area(long[] heights) {
        int i = 0, j = heights.Length - 1;
        long best = 0;
        while (i < j) {
            long area = Math.Min(heights[i], heights[j]) * (j - i);
            best = Math.Max(best, area);
            if (heights[i] < heights[j]) i++;
            else j--;
        }
        return best;
    }
}

File: internal/content/testdata/bench/rust/nc-container-most-water.rs
fn max_area(heights: Vec<i64>) -> i64 {
    let mut i = 0i64;
    let mut j = heights.len() as i64 - 1;
    let mut best = 0i64;
    while i < j {
        let area = heights[i as usize].min(heights[j as usize]) * (j - i);
        best = best.max(area);
        if heights[i as usize] < heights[j as usize] {
            i += 1;
        } else {
            j -= 1;
        }
    }
    best
}
